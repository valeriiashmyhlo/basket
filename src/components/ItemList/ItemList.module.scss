.itemList {
  display: flex;
  justify-content: center;
  flex-flow: wrap;
};
