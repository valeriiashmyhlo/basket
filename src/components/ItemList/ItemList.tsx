import * as React from "react";
import { Item as ItemComponent } from "../Item";

interface Item {
  name: string;
  price: number;
}

interface ItemListProps {
  itemList: Item[];
}

const ItemList: React.FC<ItemListProps> = ({ itemList }) => {
  const onAddToCart = () => {};
  return (
    <div>
      {itemList.map((item, i) => (
        <ItemComponent key={i} onAddToCart={onAddToCart} {...item} />
      ))}
    </div>
  );
};

export { ItemList };
