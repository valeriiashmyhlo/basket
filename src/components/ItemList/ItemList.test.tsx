import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { ItemList } from "./ItemList";

describe(`<ItemList /> tests`, () => {
  it("should render ItemList with correct props", () => {
    const items = [
      {
        name: "Face Mask",
        price: 2.5,
      },
      {
        name: "Toilet Paper",
        price: 0.65,
      },
      {
        name: "Hand Sanitizer",
        price: 19.99,
      },
    ];
    const { container } = render(<ItemList itemList={items} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
