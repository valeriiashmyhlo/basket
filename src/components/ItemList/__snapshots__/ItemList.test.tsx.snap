// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemList /> tests should render ItemList with correct props 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card-title h5"
    >
      Face Mask
    </div>
    <div
      class="card-body"
    >
      2.5
    </div>
    <button
      class="btn btn-primary"
      type="button"
    >
      Add to cart
    </button>
  </div>
  <div
    class="card"
  >
    <div
      class="card-title h5"
    >
      Toilet Paper
    </div>
    <div
      class="card-body"
    >
      0.65
    </div>
    <button
      class="btn btn-primary"
      type="button"
    >
      Add to cart
    </button>
  </div>
  <div
    class="card"
  >
    <div
      class="card-title h5"
    >
      Hand Sanitizer
    </div>
    <div
      class="card-body"
    >
      19.99
    </div>
    <button
      class="btn btn-primary"
      type="button"
    >
      Add to cart
    </button>
  </div>
</div>
`;
